lib/Backend/LowerMDShared.cpp

```diff
@@ -2458,8 +2458,8 @@ LowererMD::GenerateFastBrOrCmString(IR::Instr* instr)
         !srcReg2 ||
         srcReg1->IsTaggedInt() ||
         srcReg2->IsTaggedInt() ||
-        !srcReg1->GetValueType().IsLikelyString() ||
-        !srcReg2->GetValueType().IsLikelyString())
+        !srcReg1->GetValueType().HasHadStringTag() ||
+        !srcReg2->GetValueType().HasHadStringTag())
```

Comment:
Nope, good catch. I did ARM change after my earlier testing and forgot the quirk of this flag. I'll change that to use HasHadStringTag as well.